tests/core/test_library.py

```diff
@@ -206,161 +191,205 @@ def test_refresh_without_uri_calls_all_backends(self):
-            query=dict(any=['a']), uris=['dummy2:'], exact=True)
+            query={'any': ['a']}, uris=['dummy2:'], exact=False)
 
-    def test_find_exact_filters_out_none(self):
+    def test_search_filters_out_none(self):
         track1 = Track(uri='dummy1:a')
         result1 = SearchResult(tracks=[track1])
 
-        self.library1.search.return_value.get.return_value = result1
-        self.library2.search.return_value.get.return_value = None
+        self.library1.search().get.return_value = result1
+        self.library1.search.reset_mock()
+        self.library2.search().get.return_value = None
+        self.library2.search.reset_mock()
 
-        result = self.core.library.find_exact(any=['a'])
+        result = self.core.library.search({'any': ['a']})
 
         self.assertIn(result1, result)
         self.assertNotIn(None, result)
         self.library1.search.assert_called_once_with(
-            query=dict(any=['a']), uris=None, exact=True)
+            query={'any': ['a']}, uris=None, exact=False)
         self.library2.search.assert_called_once_with(
-            query=dict(any=['a']), uris=None, exact=True)
+            query={'any': ['a']}, uris=None, exact=False)
 
-    def test_find_accepts_query_dict_instead_of_kwargs(self):
+    def test_search_accepts_query_dict_instead_of_kwargs(self):
         track1 = Track(uri='dummy1:a')
         track2 = Track(uri='dummy2:a')
         result1 = SearchResult(tracks=[track1])
         result2 = SearchResult(tracks=[track2])
 
-        self.library1.search.return_value.get.return_value = result1
-        self.library2.search.return_value.get.return_value = result2
+        self.library1.search().get.return_value = result1
+        self.library1.search.reset_mock()
+        self.library2.search().get.return_value = result2
+        self.library2.search.reset_mock()
 
-        result = self.core.library.find_exact(dict(any=['a']))
+        result = self.core.library.search({'any': ['a']})
 
         self.assertIn(result1, result)
         self.assertIn(result2, result)
         self.library1.search.assert_called_once_with(
-            query=dict(any=['a']), uris=None, exact=True)
+            query={'any': ['a']}, uris=None, exact=False)
         self.library2.search.assert_called_once_with(
-            query=dict(any=['a']), uris=None, exact=True)
+            query={'any': ['a']}, uris=None, exact=False)
 
-    def test_search_combines_results_from_all_backends(self):
+    def test_search_normalises_bad_queries(self):
+        self.core.library.search({'any': 'foobar'})
+        self.library1.search.assert_called_once_with(
+            query={'any': ['foobar']}, uris=None, exact=False)
+
+
+class DeprecatedFindExactCoreLibraryTest(BaseCoreLibraryTest):
+    def setUp(self):  # noqa: N802
+        super(DeprecatedFindExactCoreLibraryTest, self).setUp()
+        self._warnings_filters = warnings.filters
+        warnings.filters = warnings.filters[:]
```

Comment:
Doesn't it read better when you make a copy to 1self._warnings_filters` instead of assigning a copy to the same place, which looked at in isolation looks like a noop?